views/module/photo/gallery.pug

```diff
@@ -155,7 +155,7 @@
                             //ko if: filter.can.s()
                             //ko foreach: filter.available.s
                             label.checkbox-inline.s
-                                input(type="checkbox", data-bind="attr: {value: s}, checked: $parent.filter.disp.s")
+                                input(type="checkbox", data-bind="attr: {value: s, disabled: s === '5' && $parent.filter.disp.s().includes(s) && $parent.filter.disp.s().length === 1}, checked: $parent.filter.disp.s")
```

Comment:
@kabalin well.. I'm a little bit stuck

Maybe this expression should be exctracted somewhere, and I also don't like this '5' hardcoding

maybe
`$parent.filter.disp.s().includes(s) && $parent.filter.disp.s().length === 1`
should be replaced with
`$parent._.isEqual([s], $parent.filter.disp.s())`

Sorry for spamming

Reply:
You know, this approach makes user gallery filters work a little odd. Say if the last remaining checkbox is "Public" it becomes  disabled, but if last removed checkbox is any other, it resets to default where many checkboxes are selected.

Re your approach, to shorten template expression, you can use something like `disabled: $parent.filter.disabled().includes(s)` and have that variable defined in JS filter object. But I think this gets overcomplicated. I tested with `filterSHandle` changed as:
```
             if (_.isEmpty(val)) {
                 // If user removes last status checkbox, set public status as default
                 this.filter.disp.s([String(statuses.keys.PUBLIC)]);
-            } else {
-                this.filterChangeHandle();
             }
+
+            this.filterChangeHandle();
         },
```
(and keeping your change at line 446 in place), this seems work for me at both general and user galleries:
* the last "Public" checkbox can't be unticked if it remained the last one in any gallery
* if any other checkbox is last one and removed, then "Public" becomes active

Reply:
yeah, sure, that's much cleaner